doc/build-helpers/trivial-build-helpers.chapter.md

```diff
@@ -24,11 +41,44 @@ By default `runCommand` runs in a stdenv with no compiler environment, whereas [
     You have to create a file or directory `$out` for Nix to be able to run the builder successfully.
     :::
 
+[allowSubstitutes]: https://nixos.org/nix/manual/#adv-attr-allowSubstitutes
+[preferLocalBuild]: https://nixos.org/nix/manual/#adv-attr-preferLocalBuild
+
+::: {.example #ex-runcommandwith}
+# Invocation of `runCommandWith`
+
+```nix
+runCommandWith {
+  name = "example";
+  derivationArgs.nativeBuildInputs = [ cowsay ];
+} ''
+  cowsay > $out <<EOMOO
+  'runCommandWith' is a bit cumbersome,
+  so we have more-ergonomic wrappers.
+  EOMOO
+''
+```
+
+:::
+
+
+## `runCommand` and `runCommandCC` {#trivial-builder-runCommand}
+
+`runCommand :: String -> AttrSet -> String -> Derivation`
+`runCommandCC :: String -> AttrSet -> String -> Derivation`
+
+`runCommand name derivationArgs buildCommand` is a derivation which runs the specified `buildCommand`
+using the `stdenvNoCC` environment.  `runCommandCC` is similar but uses the default compiler environment.
+
+To minimize dependencies, `runCommandCC` should only be used when `buildCommand` needs a C compiler.
+
+The arguments `name`, `derivationArgs`, and `buildCommand` have the same types and meaning as for [`runCommandWith`](#trivial-builder-runCommandWith)
```

Comment:
It would be better to assume that the reader doesn't need to jump to `runCommandWith` first.

Reply:
```suggestion
The arguments `name`, `derivationArgs`, and `buildCommand` have the same types and meaning as for [`runCommandWith`](#trivial-builder-runCommandWith)

arguments are inherited from [`runCommandWith`](#trivial-builder-runCommandWith)
```

Also we should document the inputs of the function (arguments) following the definition list convention that i posted above.

```
### Inputs

name (String)
: name of the produced derivation
env (AttrSet)
: Additional attributes passed to underlying mkDerivation
buildCommand (String)
: Command that is executed at build time.
```